src/main/java/org/folio/circulation/infrastructure/storage/inventory/ItemRepository.java

```diff
@@ -377,10 +377,11 @@ private ItemRepository(Clients clients, LocationRepository locationRepository,
         records.getTotalRecords()));
   }
 
-  public CompletableFuture<Result<Collection<Item>>> findByQuery(Result<CqlQuery> queryResult) {
-    FindWithCqlQuery<Item> fetcher = RecordFetching.findWithCqlQuery(itemsClient, ITEMS_COLLECTION_PROPERTY_NAME, Item::from);
+  public CompletableFuture<Result<Collection<Item>>> findByIndexNameAndHoldingsIds(String indexName, Collection<String> ids) {
+    FindWithMultipleCqlIndexValues<Item> fetcher = findWithMultipleCqlIndexValues(itemsClient,
+      ITEMS_COLLECTION_PROPERTY_NAME, Item::from);
 
-    return fetcher.findByQuery(queryResult)
+    return fetcher.find(byIndex(indexName, ids).withQuery(exactMatchAny(indexName, ids)))
```

Comment:
What does `byIndex` do? I imagine it creates a query that is overwritten by `withQuery` and so is redundant.

Reply:
This kind of fetcher uses MultipleCqlIndexValuesCriteria instance. So, its needed to convert necessary data by the mentioned static method.

Reply:
Ok, thanks for the explanation